components/brave_vpn/brave_vpn_service.cc

```diff
@@ -76,14 +168,1046 @@ std::string GetSubscriberCredentialFromJson(const std::string& json) {
+}
+
+void BraveVpnService::GetSelectedRegion(GetSelectedRegionCallback callback) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  VLOG(2) << __func__;
+
+  if (!IsValidRegion(selected_region_)) {
+    // Gives device region if there is no cached selected region.
+    VLOG(2) << __func__ << " : give device region instead.";
+    std::move(callback).Run(device_region_.Clone());
+    return;
+  }
+
+  VLOG(2) << __func__ << " : Give " << selected_region_.name_pretty;
+  std::move(callback).Run(selected_region_.Clone());
+}
+
+void BraveVpnService::SetSelectedRegion(
+    brave_vpn::mojom::RegionPtr region_ptr) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  if (connection_state_ == ConnectionState::DISCONNECTING ||
+      connection_state_ == ConnectionState::CONNECTING) {
+    VLOG(2) << __func__
+            << ": Current state: " << GetStringFor(connection_state_)
+            << " : prevent changing selected region while previous operation "
+               "is in-progress";
+    return;
+  }
+
+  VLOG(2) << __func__ << " : " << region_ptr->name_pretty;
+
+  selected_region_ = *region_ptr.Clone();
+  SetRegionToPrefs(brave_vpn::prefs::kBraveVPNSelectedRegion, selected_region_);
+
+  connection_info_.Reset();
+}
+
+void BraveVpnService::GetProductUrls(GetProductUrlsCallback callback) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+  std::move(callback).Run(brave_vpn::mojom::ProductUrls::New(
+      brave_vpn::kFeedbackUrl, brave_vpn::kAboutUrl,
+      brave_vpn::GetManageUrl()));
+}
+
+void BraveVpnService::CreateSupportTicket(
+    const std::string& email,
+    const std::string& subject,
+    const std::string& body,
+    CreateSupportTicketCallback callback) {
+  auto internal_callback =
+      base::BindOnce(&BraveVpnService::OnCreateSupportTicket,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+
+  const GURL base_url = GetURLWithPath(kVpnHost, kCreateSupportTicket);
+  base::Value dict(base::Value::Type::DICTIONARY);
+
+  std::string email_trimmed, subject_trimmed, body_trimmed, body_encoded;
+  base::TrimWhitespaceASCII(email, base::TRIM_ALL, &email_trimmed);
+  base::TrimWhitespaceASCII(subject, base::TRIM_ALL, &subject_trimmed);
+  base::TrimWhitespaceASCII(body, base::TRIM_ALL, &body_trimmed);
+  base::Base64Encode(body_trimmed, &body_encoded);
+
+  // required fields
+  dict.SetStringKey("email", email_trimmed);
+  dict.SetStringKey("subject", subject_trimmed);
+  dict.SetStringKey("support-ticket", body_encoded);
+  dict.SetStringKey("partner-client-id", "com.brave.browser");
+
+  // optional (but encouraged) fields
+  dict.SetStringKey("subscriber-credential", "");
+  dict.SetStringKey("payment-validation-method", "brave-premium");
+  dict.SetStringKey("payment-validation-data", "");
+
+  std::string request_body = CreateJSONRequestBody(dict);
+  VLOG(2) << " : sending " << request_body;
+  OAuthRequest(base_url, "POST", request_body, std::move(internal_callback));
```

Comment:
i assume these are not sending/saving cookies due to https://github.com/brave/brave-core/blob/837a78c6921319228771eb94921d6f7ea89c4e00/components/api_request_helper/api_request_helper.cc#L63-L65 but wanted to double check

also i was wondering if we should be setting a cookie policy in the network traffic annotation tag, like so: https://github.com/brave/brave-core/blob/1cb5818aa0b70666c6aeea5ea9c06cc4e712171a/components/brave_wayback_machine/wayback_machine_url_fetcher.cc#L39-L43 (in https://github.com/brave/brave-core/blob/5b576ef6125c629d3c876f335a9927f7ca5c208d/components/brave_vpn/brave_vpn_service.cc#L28)

Reply:
No cookies sent (or stored!). Let me see if I can update the annotation

Reply:
Already done anotation updating :)  - https://github.com/brave/brave-core/pull/12175/commits/ed20ecca5fe13073cd60bf625de53a05d7b4c098